spec/requests/order_request_spec.rb

```diff
@@ -0,0 +1,292 @@
+      before {get :new, params: {user_id: user.id,
+                                 date_start: "2020-12-21",
+                                 date_end: "2020-12-22"}}
+
+      it "should redirect to root path" do
+        expect(response).to redirect_to root_path
+      end
+    end
+
+    context "when empty param date_start" do
+      before {get :new, params: {user_id: user.id,
+                                 room_id: room.id,
+                                 date_end: "2020-12-22"}}
+
+      it "should have date_start equal today" do
+        expect(assigns(:date_start)).to eq Time.zone.today.to_s
+      end
+    end
+  end
+
+  describe "POST #create" do
+    context "when valid param" do
+      before {post :create, params: {user_id: user.id,
+                                     room_id: room.id,
+                                     date_start: "2020-12-21",
+                                     date_end: "2020-12-22",
+                                     order: {quantity_person: 1,
+                                             date_start: "2020-12-21",
+                                             date_end: "2020-12-22"}}}
+
+      it "should create new order" do
+        expect{
+          post :create, params: {user_id: user.id,
+                                 room_id: room.id,
+                                 order: params_order.merge(order: {quantity_person: 1,
+                                                                   date_start: "2020-12-21",
+                                                                   date_end: "2020-12-22"})}
+        }.to change(Order, :count).by 1
+      end
+
+      it "should redirect to rooth path" do
+        expect(response).to redirect_to root_path
+      end
+    end
+
+    context "when invalid param" do
+      before {post :create, params: {user_id: user.id,
+                                     room_id: room.id,
+                                     date_start: "2020-12-21",
+                                     date_end: "2020-12-22",
+                                     order: {quantity_person: 1,
+                                             date_start: "2020-12-21",
+                                             date_end: "2020-12-22"}}}
+
+      it "should not create new order" do
+        expect{
+          post :create, params: {user_id: user.id,
+                                 room_id: room.id,
+                                 order: invalid_params_order.merge(order: {quantity_person: 1,
+                                                                           date_start: "2020-12-21",
+                                                                           date_end: "2020-12-22"})}
+        }.to change(Order, :count).by 0
+      end
+
+      it "should redirect to rooth path" do
+        expect(response).to redirect_to root_path
+      end
+
+      context "when quantity people more than allow" do
+        before(:each) do
```

Comment:
before each là sao e

Reply:
before each để trước mỗi khối it đều định nghĩa cái request referer kia ạ